### PR TITLE
Added NanTryCatch

### PR DESCRIPTION
**Pull request by kkoopa**
`TryCatch` will start requiring an `Isolate *`.

### CODE REVIEW

File: test/cpp/trycatch.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+void Init (v8::Handle<v8::Object> target) {
+  target->Set(
+      NanNew<v8::String>("r")
+    , NanNew<v8::FunctionTemplate>(TryCatch)->GetFunction()
+  );
+}
+
+NODE_MODULE(trycatch, Init)
+
```
Comment (bnoordhuis) [inline test/cpp/trycatch.cpp]: Blank line at end.

File: test/js/trycatch-test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ * NAN - Native Abstractions for Node.js
+ *
+ * Copyright (c) 2015 NAN contributors
+ *
+ * MIT License <https://github.com/nodejs/nan/blob/master/LICENSE.md>
+ ********************************************************************/
+
+const test     = require('tap').test
+    , testRoot = require('path').resolve(__dirname, '..')
+    , bindings = require('bindings')({ module_root: testRoot, bindings: 'trycatch' });
```
Comment (bnoordhuis) [inline test/js/trycatch-test.js]: Long line?

Comment (kkoopa) [inline test/js/trycatch-test.js]: It's like that in all javascript.

File: test/cpp/trycatch.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ * MIT License <https://github.com/nodejs/nan/blob/master/LICENSE.md>
+ ********************************************************************/
+
+#include <nan.h>
+
+NAN_METHOD(TryCatch) {
+  NanTryCatch try_catch;
+  v8::Local<v8::Value> v =
+      NanRunScript(NanNew<NanUnboundScript>(NanNew("throw 'waaa'")));
+  if (v.IsEmpty()) {
```
Comment (bnoordhuis) [inline test/cpp/trycatch.cpp]: Assert that `try_catch.HasCaught()` here?

**Comment by bnoordhuis** [comment]
LGTM with suggestions.

**Comment by kkoopa** [comment]
Fixed all but the long line in js, there is no linting of that code and it is not conformant to any sort of style. Cleaning all of that is better left for a separate PR. Will merge now. Thanks.

**Pull request merged**